src/tests/utils.ts

```diff
@@ -45,12 +47,53 @@ export async function queryNamespaces(
   const orgNsPromise = connection.query(orgNsQuery);
 
   const allNamespaces = await Promise.all([installedNsPromise, orgNsPromise]);
-  const installedNamespaces = allNamespaces[0].records.map((record) => {
+  const installedNamespaces = allNamespaces[0].records.map(record => {
     return { installedNs: true, namespace: record.NamespacePrefix };
   });
-  const orgNamespaces = allNamespaces[1].records.map((record) => {
+  const orgNamespaces = allNamespaces[1].records.map(record => {
     return { installedNs: false, namespace: record.NamespacePrefix };
   });
 
   return [...orgNamespaces, ...installedNamespaces];
 }
+
+export function verifyCountQueries(
+  countQueryResult: { expr0: number }[],
+  countQueries: string[]
+) {
+  // check count query results to ensure that there are results that can be used to set maxFetch
+  countQueryResult.forEach((result, index) => {
+    if (!result?.expr0 || result?.expr0 <= 0) {
+      throw new Error(
+        nls.localize(
+          'invalidCountQueryResult',
+          countQueries[index].substring(
+            0,
+            countQueries[index].indexOf('IN (') + 30
+          )
+        )
+      );
+    }
+  });
+}
+
+export const queryAll = async <T>(
+  connection: Connection,
+  query: string,
+  tooling = false
+): Promise<QueryResult<T>> => {
+  const conn = tooling ? connection.tooling : connection;
+  const allRecords: T[] = [];
+  let result = await conn.query<T>(query);
+  allRecords.push(...result.records);
+  while (!result.done) {
+    result = (await conn.queryMore(result.nextRecordsUrl)) as QueryResult<T>;
+    allRecords.push(...result.records);
+  }
+
+  return {
+    done: true,
+    totalSize: allRecords.length,
```

Comment:
can we log these details (totalSize) during the run?

Reply:
@diyer do you mean so they can be seen in telemetry or just in the typical local log files for the user?

Reply:
log files for the user